### DISCUSSION TITLE
How to add NAS shares to PLEX library?

### DISCUSSION THREAD

**Discussion by eili77** [tteck/Proxmox#143] [Q&A]
Hi, a newbie question:
I installed PLEX with the script provided on ProxMox.
I also got a QNAP NAS in my network with all my media files. Shares are defined to access the content.
What would I have to do to be able to use these shares when setting up libraries in PLEX?

Many thanks in advance!

**Comment by AKHwyJunkie** [comment]
Doing this in Proxmox is a bit tricky.  I did it with a manual build, but these scripts will work exactly the same.  These are the overall steps, missing a lot of detail of course.

1. You need a priviledged container as you'll need it to do the file shares.  tteck's new v3 scripts can do this or you can do it via backup/restore, but there's no way to "convert" an existing container if it was built as unpriviledged.
2. The container will need the CIFS or NFS features enabled, depending on what you'll use. (LXC->Options->Features)   Personally, I built out my file shares with CIFS as that's my NAS' native protocol.
3. You will likely need to install cifs-utils on the Plex server, if you go the CIFS route.
4. From there, you need to mount the share on the Plex server, procedure/command is slightly different depending on whether it's CIFS or NFS and whether you're using authentication.
5. Add the mounted share to /etc/fstab in the Plex LXC for permanent mounting.
6. Add the mounted file share as a directory within Plex

I really liked disconnecting Plex from my NAS.  It offered me the ability to have a 24/7 music directory and I only have to turn on my NAS for video and my "deep" music archives.  I do "sometimes" get weird CIFS availability issues with the Plex server always-on and the NAS not being so.  Rebooting the Plex LXC solves it every time.
